README.md

```diff
@@ -212,6 +212,69 @@ The alt attribute functions as an alternative source of inclusion should the url
 <h-include src="unavailable.html" alt="alternative.html"></h-include>
 ```
 
+### HTTP Headers
+
+Set custom request headers and subscribe to response headers by using the HIncludeConfig configuration, example:
+
+```js
+HIncludeConfig = {
+  setRequestHeaders: [{key, value}],
+  getResponseHeaderKeys: ['key'],
```

Comment:
So, to my understanding, the client provides a list of response header names to act on and then a required callback when those response headers are seen.

I think that if we instead used something similar to the [EventListener](https://developer.mozilla.org/en-US/docs/Web/API/EventListener) interface, we would increase cohesion and also reduce complexity in the  client callbacks.

The downside would be logic that requires multiple headers being present would be harder to write. But this can be mitigated by keeping your proposed callback parameter, still providing the full list of response headers.